src/main/java/com/algolia/search/util/AlgoliaUtils.java

```diff
@@ -0,0 +1,167 @@
+package com.algolia.search.util;
+
+import com.algolia.search.exceptions.AlgoliaRuntimeException;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+import org.apache.http.HttpResponse;
+import org.apache.http.concurrent.FutureCallback;
+
+public class AlgoliaUtils {
+
+  /** Checks if the given string is empty or white spaces */
+  public static Boolean isEmptyWhiteSpace(final String stringToCheck) {
+    return stringToCheck.trim().length() == 0;
+  }
+
+  /** Checks if the given string is null, empty or white spaces */
+  public static Boolean isNullOrEmptyWhiteSpace(final String stringToCheck) {
+    return stringToCheck == null || stringToCheck.trim().length() == 0;
+  }
+
+  /**
+   * Ensure that the objectID field or the @JsonProperty(\"objectID\")" is present in the given
+   * class
+   *
+   * @param clazz The class to scan
+   * @throws AlgoliaRuntimeException When the class doesn't have an objectID field or a Jackson
+   *     annotation @JsonProperty(\"objectID\"")
+   */
+  public static <T> void ensureObjectID(@Nonnull Class<T> clazz) {
```

Comment:
Seems to duplicate the logic of `getObjectID`, which could be leveraged here :upside_down_face:

Reply:
I'll have a look